### ISSUE TITLE
Label tensor in MultipleNegativesRankingLoss

### ISSUE THREAD

**Issue by jamescalam** [UKPLab/sentence-transformers#1204]
Hi I'm hoping this isn't too stupid a question! I'm trying to make sure I am correctly understanding the multiple negatives ranking loss function and I thought I was mostly there, but there's one thing in the code that doesn't quite make sense. In [MultipleNegativesRankingLoss.py](https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/losses/MultipleNegativesRankingLoss.py) I see that the `labels` tensor is calculated as:

```python
labels = torch.tensor(range(len(scores)), dtype=torch.long, device=scores.device)
```

When I test this, I return a single tensor value `tensor([0])`, which to my understanding makes sense as we'd want the cosine similarity to be highest for `scores` at position `0`.

My question here is either, do I have something wrong in my code (it shouldn't be `0`) - or where would `labels` not be `0`? Eg why can we not write `labels = torch.tensor([0])`?

Thanks a lot!

**Comment by nreimers** [comment]
@jamescalam Yes, that is correct. 

In this talk I also provide an explanation of MultipleNegativesRankingLoss:
https://www.youtube.com/watch?v=XHY-3FzaLGc
